looper/looper.py

```diff
@@ -555,20 +566,700 @@ def __call__(self, args, remaining_args):
+                while num_figures < 3:
+                    # Add additional empty columns for clean format
+                    obj_figs.append("\t\t\t  <div class='col'>")
+                    obj_figs.append("\t\t\t  </div>")
+                    num_figures += 1
+                return ("\n".join(["\t\t<h5>Looper project objects</h5>",
+                                   "\t\t<div class='container'>",
+                                   "\t\t\t<div class='row justify-content-start'>",
+                                   "\n".join(obj_figs),
+                                   "\t\t\t</div>",
+                                   "\t\t</div>",
+                                   OBJECTS_LIST_HEADER,
+                                   "\n".join(obj_links),
+                                   OBJECTS_LIST_FOOTER]))
+
+        def create_index_html(objs, stats):
+            # Generate an index.html style project home page w/ sample summary
+            # statistics
+            objs.drop_duplicates(keep='last', inplace=True)
+            reports_dir = os.path.join(self.prj.metadata.output_dir, "reports")
+            # Generate parent index.html page
+            objs_html_path = "{root}_summary.html".format(
+                root=os.path.join(self.prj.metadata.output_dir, self.prj.name))
+            # Generate parent objects.html page
+            object_parent_path = os.path.join(reports_dir, "objects.html")
+            # Generate parent samples.html page
+            sample_parent_path = os.path.join(reports_dir, "samples.html")
+
+            objs_html_file = open(objs_html_path, 'w')
+            objs_html_file.write(HTML_HEAD_OPEN)
+            objs_html_file.write("\t\t<style>\n")
+            objs_html_file.write(TABLE_STYLE_ROTATED_HEADER)
+            objs_html_file.write(TABLE_STYLE_TEXT)
+            objs_html_file.write("\t\t</style>\n")
+            objs_html_file.write(HTML_TITLE.format(project_name=self.prj.name))
+            navbar = create_navbar(objs, self.prj.metadata.output_dir)
+            objs_html_file.write(navbar)
+            objs_html_file.write(HTML_HEAD_CLOSE)
+
+            # Add stats_summary.tsv button link
+            stats_relpath = os.path.relpath(tsv_outfile_path,
+                                            self.prj.metadata.output_dir)
+            objs_html_file.write(HTML_BUTTON.format(
+                file_path=stats_relpath, label="Stats Summary File"))
+
+            # Add stats summary table to index page
+            if os.path.isfile(stats_file):
+                objs_html_file.write(TABLE_HEADER)
+                # Produce table columns
+                sample_pos = 0
+                # Get unique column name list
+                col_names = []
+                while sample_pos < len(stats):
+                    for key, value in stats[sample_pos].items():
+                        col_names.append(key)
+                    sample_pos += 1
+                unique_columns = []
```

Comment:
Building this looks like a good candidate for `uniqify` like in `pypiper.utils`?

Reply:
probably good to duplicate that function here (not introduce a dependency on pypiper)

Reply:
Added uniqify and implemented that as replacement (713527d).

Reply:
Cool, yeah not worth the dependency but nice to have within this project for simplicity/reuse